docker/entrypoint.sh

```diff
@@ -75,8 +75,10 @@ case ${entry_cmd} in
 	    case ${test_py} in
```

Comment:
Aside: is there any reason why we run tests in our Docker entrypoint? Is this needed (given we have CI)? Perhaps we should consider removing @francbartoli @justb4

Reply:
I am working on it already on it on the other PR, as the entrypoint should not have commands, but just prepare the environment.

Reply:
I think we should be able to run (most of) the Unit-tests within a Docker Container. Especially currently, the entire installation with OS, Python and package-versions will surely differ from a custom install. Hopefully this will largely go away with PDM and #1105, but still, just relying on `main.yml` is fragile or we should somehow invoke there. Ok, maybe via `entrypoint.sh` is not the cleanest way but we will also need some logic to skip tests.

Reply:
And btw the above Docker build does not even invoke Unit tests. There is some larger problem there. After 6 hours the above build just was canceled, think by GH, no logs.

Reply:
Also containers.yml is now running for 5 hours after PR #1109 (unrelated to Docker build). At least the image is still built twice so that should be removed ASAP.

Reply:
How does it take to build locally @justb4?

Reply:
Takes around 8 minutes on my system.

Reply:
And the ran the tests:  `docker run -p 5000:80 --rm --name pygeoapi geopython/pygeoapi:latest test > test.txt 2>&1`. Output attached:
[test.txt](https://github.com/geopython/pygeoapi/files/10457130/test.txt)

Actually only 3 main failure-causes, most can be explained:

* several tests in [test_api.py](https://github.com/geopython/pygeoapi/blob/master/tests/test_api.py)
Many tests IMO should not be here! Like tests for Postgresql CQL,Coverage, EDR. They should be in their Provider (PG) or Service-specific testfile.  `test_api.py` should perform generic API tests, not specific, and relies on unavailable local servers.

* WFS tests: still the wrong GeoSolutions URL is in, thought we fixed that ages ago...
* `test_async_hello_world_process_parallel` fails, cannot explain

All other tests succeed, that is the good news, but still we should open an issue to cleanup tests.

Reply:
Takes less than 3 minutes on mine

Reply:
Opened issue: #1111.